### PR TITLE
connect shared modules using build dependencies

### PR DESCRIPTION
**Pull request by zvictor**
based on `build.dependencies` from https://docs.garden.io/guides/configuration-files#module-configuration

image

### CODE REVIEW

File: src/backend/Dockerfile [hunk 1]
```diff
@@ -4,43 +4,40 @@
 FROM node:alpine AS base
 RUN apk add --no-cache nodejs-current tini
 ENV HOME=/home/node
-USER node
+# USER node
```
Comment (zvictor) [inline src/backend/Dockerfile]: permissions break when we run `npm install` over the new files shared by garden

File: src/backend/Dockerfile [hunk 2]
```diff
@@ -4,43 +4,40 @@
...
 WORKDIR $HOME/backend
-COPY backend/package*.json backend/setup-npm-env.sh ./
+COPY package*.json ./
 RUN touch config.js && \
-	npm set progress=false && \
-    npm config set depth 0 && \
-    npm install --only=production
+    npm set progress=false && \
+    npm config set depth 0
+RUN npm install --only=production
```
Comment (zvictor) [inline src/backend/Dockerfile]: split for better caching

File: src/backend/Dockerfile [hunk 3]
```diff
@@ -4,43 +4,40 @@
...
 
 ##
 ## Test dependencies
 ##
 FROM production-dependencies AS test-dependencies
-RUN npm set progress=false && \
-    npm config set depth 0 && \
-    npm install
+RUN npm install
```
Comment (zvictor) [inline src/backend/Dockerfile]: was there a reason to run the same set of commands twice?

**Comment by jpsecher** [comment]
Squashed into master.

**Pull request closed without merging**